app/scripts/metamask-controller.js

```diff
@@ -588,7 +598,13 @@ export default class MetamaskController extends EventEmitter {
       getERC20TokenName: this.assetsContractController.getERC20TokenName.bind(
         this.assetsContractController,
       ),
-      config: { provider: this.provider },
+      config: {
+        provider: this.provider,
+        selectedAddress:
+          initState?.AccountsController?.internalAccounts?.accounts[
+            initState?.AccountsController?.internalAccounts?.selectedAccount
+          ]?.address ?? '',
```

Comment:
Nit: It seems `initState` is always set — do you need the `?.`s after it?

``` suggestion
          initState.AccountsController?.internalAccounts?.accounts[
            initState.AccountsController?.internalAccounts?.selectedAccount
          ]?.address ?? '',

Reply:
Solved in [f703fae](https://github.com/MetaMask/metamask-extension/pull/21437/commits/f703fae5d788278a1f889122b0bbb175e66c3b07)